resources/js/test/setup.tsx

```diff
@@ -60,3 +60,37 @@ export function render<TPageProps = Record<string, unknown>>(
     ...defaultRender(ui, { wrapper, ...options }),
   };
 }
+
+// -----------------
+
+type DefaultRenderHookParams = Parameters<typeof defaultRenderHook>;
+type RenderCallback = DefaultRenderHookParams[0];
+type RenderHookOptions<TPageProps = Record<string, unknown>> = DefaultRenderHookParams[1] & {
+  pageProps?: Partial<TPageProps & AppGlobalProps>;
+};
+
+export function renderHook<TPageProps = Record<string, unknown>>(
```

Comment:
I really don't like ever using `renderHook()` to test React hooks without an accompanying component that they influence (ie: test a component that just happens to use the hook for behavior rather than explicitly testing the hook itself).

But some of the hooks with the table are complex enough that I felt it was warranted.